internal/shellcmd/mode.go

```diff
@@ -2,20 +2,43 @@ package shellcmd
 
 import (
 	"fmt"
+	"strings"
 
 	"github.com/libsql/libsql-shell-go/pkg/shell/enums"
 	"github.com/spf13/cobra"
 )
 
+func validArg(provided string, valid []string) bool {
+	for i := 0; i < len(valid); i++ {
+		if provided == valid[i] {
+			return true
+		}
+	}
+	return false
+}
+
 var modeCmd = &cobra.Command{
 	Use:   ".mode MODE",
 	Short: "Set output mode",
-	Args:  cobra.ExactArgs(1),
+	Args:  cobra.MaximumNArgs(1),
+	ValidArgs: []string{
+		string(enums.TABLE_MODE),
+		string(enums.JSON_MODE),
+		string(enums.CSV_MODE),
+	},
 	RunE: func(cmd *cobra.Command, args []string) error {
+		validModes := strings.Join(cmd.ValidArgs, ", ")
 		config, ok := cmd.Context().Value(dbCtx{}).(*DbCmdConfig)
 		if !ok {
 			return fmt.Errorf("missing db connection")
 		}
+		currentMode := config.GetMode()
+		if len(args) == 0 {
+			return fmt.Errorf("No mode provided. Current mode is %s. Valid modes are %s", currentMode, validModes)
+		}
+		if !validArg(args[0], cmd.ValidArgs) {
+			return fmt.Errorf("Invalid mode. Current mode is %s. Valid modes are %s", currentMode, validModes)
+		}
```

Comment:
No need to this `if`. Switch's `default` case is exactly this. You feel free to use the message that you this it's best

Reply:
True! Do we plan on supporting more modes soon? Might be a good idea to get rid of the switch statement in favor of something that scales better. Sqlite3 has like 14 modes.

Reply:
I don't think you can get rid of that complexity, just move to another place. Golang switch statements are fine

Reply:
that is very readable the way it is right now so it is ok, we can abstract after it gets too cluttered